solr/core/src/java/org/apache/solr/cloud/gumi/ReplicaType.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.cloud.gumi;
+
+/**
+ * TODO: move into {@link Replica}?
+ */
+public enum ReplicaType {
```

Comment:
Why not reuse the existing Replica.Type enum ?

Reply:
I didn't want to expose `Replica` as stated above and I therefore can't use `Replica.Type`.
If `Type` were in its own Java file and a pure enum, reuse would have been possible. But even then, not sure it would have been a good idea: an internal replica type might eventually include things we don't want a plugin to be able to manipulate.